packages/running-extension/src/kernels.tsx

```diff
@@ -41,14 +45,81 @@ export async function addKernelRunningSessionManager(
   const { runningChanged, RunningKernel } = Private;
   const throttler = new Throttler(() => runningChanged.emit(undefined), 100);
   const trans = translator.load('jupyterlab');
+  const shutdownUnusedLabel = trans.__('Shut Down Unused');
+  let shutdownUnusedEnabled = false;
+  const shutdownUnusedThrottler = new Throttler(
+    checkShutdownUnusedEnabled,
+    5000
+  );
 
   // Throttle signal emissions from the kernel and session managers.
-  kernels.runningChanged.connect(() => void throttler.invoke());
+  kernels.runningChanged.connect(() => {
+    void throttler.invoke();
+    void shutdownUnusedThrottler.invoke();
+  });
   sessions.runningChanged.connect(() => void throttler.invoke());
 
   // Wait until the relevant services are ready.
   await Promise.all([kernels.ready, kernelspecs.ready, sessions.ready]);
 
+  async function getUnusedKernels() {
+    // Identifies unused kernels
+    const runningKernels = await KernelAPI.listRunning();
+    return runningKernels.filter(
+      kernel => kernel.connections !== undefined && kernel.connections < 1
+    );
```

Comment:
Using the following code works for me:

```suggestion
  function getUnusedKernels() {
    // Identifies unused kernels
    return Array.from(kernels.running()).filter(
      kernel => (kernel.connections ?? 1) < 1
    );
```